### PR TITLE
Fix SOS clrstack w^x issues on osx-arm64

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

File: src/coreclr/vm/dllimportcallback.cpp [hunk 1]
```diff
@@ -1098,6 +1101,10 @@ PCODE TheUMEntryPrestubWorker(UMEntryThunk * pUMEntryThunk)
     if (pThread->IsAbortRequested())
         pThread->HandleThreadAbort();
 
+#if defined(HOST_OSX) && defined(HOST_ARM64)
+        auto jitWriteEnableHolder = PAL_JITWriteEnable(true);
```
Comment (janvorli) [inline src/coreclr/vm/dllimportcallback.cpp:1105]: A nit - wrong indentation

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/coreclr/vm/dllimportcallback.cpp (hunk 1 above)
Comment (sdmaclea) [reply src/coreclr/vm/dllimportcallback.cpp:1105]: Someone should really fix cut and paste.  The number of times the clipboard created errors in the last decades. 😉

**Comment by sdmaclea** [comment]
For future reference these were the callstack for the W^X errors occurred.
```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libcoreclr.dylib                       0x00000001145851a4 UMEntryThunkCache::GetUMEntryThunk(MethodDesc*) + 364
1   libcoreclr.dylib                       0x000000011458519c UMEntryThunkCache::GetUMEntryThunk(MethodDesc*) + 356
2   libcoreclr.dylib                       0x000000011448d6d0 CorHost2::CreateDelegate(unsigned int, char16_t const*, char16_t const*, char16_t const*, long*) + 800
3   libcoreclr.dylib                       0x00000001144408d8 coreclr_create_delegate + 112
4   libsos.dylib                           0x000000010f5e1894 InitializeHosting() + 3048
```

```
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address
=0x311294188)                                                                              
  * frame #0: 0x00000001117004d8 libcoreclr.dylib`InterlockedCompareExchange64 + 36       
    frame #1: 0x0000000111833758 libcoreclr.dylib`UtilCode::InterlockedCompareExchangeHelpe
r<unsigned long, 8>::InterlockedCompareExchange(unsigned long volatile*, unsigned long, uns
igned long) + 48                             
    frame #2: 0x000000011182bfd4 libcoreclr.dylib`unsigned long InterlockedCompareExchangeT
<unsigned long>(unsigned long volatile*, unsigned long, unsigned long) + 60
    frame #3: 0x0000000111a54f74 libcoreclr.dylib`UMThunkMarshInfo::RunTimeInit() + 392
    frame #4: 0x0000000111a54710 libcoreclr.dylib`UMEntryThunk::RunTimeInit() + 44
    frame #5: 0x0000000111a544d4 libcoreclr.dylib`UMEntryThunk::DoRunTimeInit(UMEntryThunk*
) + 128                                      
    frame #6: 0x0000000111a543fc libcoreclr.dylib`TheUMEntryPrestubWorker + 132
    frame #7: 0x0000000111d0faf4 libcoreclr.dylib`TheUMEntryPrestub + 52
```

**Comment by janvorli** [comment]
Thanks for the call stacks Steve. These both make sense. I am surprised though that we haven't hit them in the coreclr tests.

**Comment by sdmaclea** [comment]
Ack. We may hit a few more when we run the corefx and diagnostics tests

**Comment by janvorli** [comment]
Maybe it is just hidden behind other issues in coreclr tests.

**Pull request merged**